glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -474,4 +476,27 @@ class GleanTest {
         Glean.setUploadEnabled(true)
         assertTrue(GleanInternalMetrics.os.testHasValue())
     }
+
+    @Test
+    fun `Workers should be cancelled when disabling uploading`() {
+        // Force the MetricsPingScheduler to schedule the MetricsPingWorker
+        Glean.metricsPingScheduler.schedulePingCollection(Calendar.getInstance(), true)
+        // Enqueue a worker to send the baseline ping
+        Pings.baseline.send()
+
+        // Verify that the workers are enqueued
+        assertTrue("PingUploadWorker is enqueued",
+            getWorkerStatus(PingUploadWorker.PING_WORKER_TAG).isEnqueued)
+        assertTrue("MetricsPingWorker is enqueued",
+            getWorkerStatus(MetricsPingWorker.TAG).isEnqueued)
+
+        // Toggle upload enabled to false
+        Glean.setUploadEnabled(false)
```

Comment:
We also need to add a test for making sure that we don't cancel workers when flipping things on.